File: backend/src/main/java/com/back/domain/item/item/dto/MostReplacedItemResponse.java
package com.back.domain.item.item.dto;

import java.util.List;
import java.util.Map;

public record MostReplacedItemResponse(
        Long itemId,
        String itemName,
        String categoryName,
        Long replacementCount,
        String imgUrl
) {
    /**
     * Repository 쿼리 결과(Map)를 DTO로 변환
     *
     * @param result Repository에서 반환한 Map 데이터
     * @return MostReplacedItemResponse DTO
     */
    public static MostReplacedItemResponse from(Map<String, Object> result) {
        return new MostReplacedItemResponse(
                ((Number) result.get("itemId")).longValue(),
                (String) result.get("itemName"),
                (String) result.get("categoryName"),
                ((Number) result.get("replacementCount")).longValue(),
                (String) result.get("imgUrl")
        );
    }

    /**
     * 여러 Map을 한번에 변환
     *
     * @param results Repository에서 반환한 Map 리스트
     * @return MostReplacedItemResponse DTO 리스트
     */
    public static List<MostReplacedItemResponse> fromList(List<Map<String, Object>> results) {
        return results.stream()
                .map(MostReplacedItemResponse::from)
                .toList();
    }
}

File: backend/src/main/java/com/back/domain/item/item/dto/MostReplacedItemResponse.kt
package com.back.domain.item.item.dto


data class MostReplacedItemResponse(
    @JvmField val itemId: Long,
    @JvmField val itemName: String,
    @JvmField val categoryName: String,
    @JvmField val replacementCount: Long,
    @JvmField val imgUrl: String?
) {
    companion object {
        /**
         * Repository 쿼리 결과(Map)를 DTO로 변환
         *
         * @param result Repository에서 반환한 Map 데이터
         * @return MostReplacedItemResponse DTO
         */
        @JvmStatic
        fun from(result: Map<String, Any?>): MostReplacedItemResponse {
            return MostReplacedItemResponse(
                itemId = (result["itemId"] as Number).toLong(),
                itemName = result["itemName"] as String,
                categoryName = result["categoryName"] as String,
                replacementCount = (result["replacementCount"] as Number).toLong(),
                imgUrl = result["imgUrl"] as? String
            )
        }

        /**
         * 여러 Map을 한번에 변환
         *
         * @param results Repository에서 반환한 Map 리스트
         * @return MostReplacedItemResponse DTO 리스트
         */
        @JvmStatic
        fun fromList(results: List<Map<String, Any?>>): List<MostReplacedItemResponse> {
            return results.map(::from)
        }
    }
}